crates/core/src/factories/core_fetch.rs

```diff
@@ -240,7 +241,7 @@ impl CoreFetch {
                     outgoing_request_tx.clone(),
                     outgoing_request_rx.clone(),
                     space_id.clone(),
-                    transport.clone(),
+                    Arc::downgrade(&transport),
```

Comment:
To test this change, I needed offline agents. The simplest way to do that was to start a test harness and shut it down. At that point, I discovered we can't shut down K2 properly. I don't know if I caught everything but this was the common issue that prevented gossip from shutting down at least.

The modules I've made this change to register themselves as module handlers, so the transport has a reference to us. We can't then have an owned reference to the transport and expect everything to drop correctly.